src/components/common/hooks/FileReader.ts

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect } from 'react'
+import { FileDataInterface, FileReaderStatusInterface, ReadFileAs, ValidatorType } from './types'
+import { FILE_READING_FAILED_STATUS, NO_FILE_SELECTED_STATUS } from './constants'
+
+export const useFileReader = () => {
+    const [fileData, setFileData] = useState<FileDataInterface>(null)
+    const [progress, setProgress] = useState<number>(0)
+    const [status, setStatus] = useState<FileReaderStatusInterface>(null)
+    // Since its a function and not a value, we need to use a callback to set the value
+    const [validator, setValidator] = useState<ValidatorType>(null)
+    const reader = new FileReader()
+    useEffect(() => {
+        return () => {
+            reader.removeEventListener('load', handleFileRead)
+            reader.removeEventListener('error', handleFileError)
+            reader.removeEventListener('progress', handleFileProgress)
+        }
+    }, [])
+
+    useEffect(() => {
+        // doing == since we want to check for null and undefined
+        if (!fileData || fileData.data == null) return
+        if (!validator) return
+        const { status, message } = validator(fileData)
+        setStatus(() => ({
+            message,
+            status,
+        }))
+        if (status) {
+            setProgress(() => 100)
```

Comment:
We can directly assign the value directly. No need to create inline method if we are not using the prev value. Handle it everyWhere

Reply:
Done